aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -2165,5 +2166,29 @@ Tensor sum_sparse_coo(const Tensor& self, at::OptionalIntArrayRef dim, bool keep
   return result;
 }
 
+Tensor sum_sparse_csr(const Tensor& self, at::OptionalIntArrayRef dim, bool keepdim, c10::optional<ScalarType> dtype) {
+  // TODO: The signature of sum.dim_IntList and _sparse_csr_sum.dim_dtype is a little
+  // bit different in the second parameters `dim`, which causes the conversion of `dim`
+  // to call into `_sparse_csr_sum`. Align the signatures would be a better choice.
+  TORCH_CHECK(
+      dim.has_value(), "dim has no value, cannot be used in sum.dim_IntList");
+  auto layout = self.layout();
+  if (layout == kSparseCsr) {
+    return at::_sparse_csr_sum(self, *dim, keepdim, dtype);
+  } else {
+    if (self.dim() != 2 || keepdim) {
+      TORCH_CHECK(
+          false,
+          "sum expected input with strided, sparse_csr layouts, got layout ",
+          layout);
+    } else if (!keepdim) {
+      TORCH_CHECK(
+          false,
+          "torch.empty: Only batched sparse compressed (non-block) tensors are supported");
+    }
```

Comment:
If we don't implement support for non-CSR layouts, there is no need to check other parameters. So:
```suggestion
    TORCH_CHECK(
        false,
        "sum expected input with strided or sparse_csr layout, got layout ",
        layout);
    }
```

Reply:
If not check the parameters, all the RuntimeError would be `sum expected input with strided or sparse_csr layout, got layout`, which could not pass the cases which expected error `torch.empty: Only batched sparse compressed`.

Reply:
The expected error `torch.empty: Only batched sparse compressed..` is just wrong and it should be updated in the sparse.py after the error message is updated.